### PR TITLE
Create only one AwsRegionalNetwork per region per benchmark run.

### PR DESCRIPTION
**Pull request by skschneider**
Store AwsRegionalNetworks in the benchmark spec.

### CODE REVIEW

File: perfkitbenchmarker/benchmark_spec.py [hunk 1]
```diff
@@ -250,7 +250,14 @@ def Prepare(self):
 
   def Provision(self):
     """Prepares the VMs and networks necessary for the benchmark to run."""
-    vm_util.RunThreaded(lambda net: net.Create(), self.networks.values())
+    # Sort networks into a guaranteed order of creation based on dict key.
+    # There is a finite limit on the number of threads that are created to
+    # provision networks. Until support is added to provision resources in an
+    # order based on dependencies, this key ordering can be used to avoid
+    # deadlock by placing dependent networks later and their dependencies
+    # earlier.
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmark_spec.py]: Thanks. Would you mind pointing out that the AWS network in particular requires this?

File: perfkitbenchmarker/providers/aws/aws_network.py [hunk 2]
```diff
@@ -410,8 +416,8 @@ def _Exists(self):
 class AwsRegionalNetwork(network.BaseNetwork):
```
Comment (cmccoy) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Should we make this class private, since creating one requires a lock?

**Comment by cmccoy** [comment]
Thanks, @skschneider. Two small comments, but LGTM.

File: perfkitbenchmarker/benchmark_spec.py (hunk 1 above)
Comment (skschneider) [inline perfkitbenchmarker/benchmark_spec.py]: Done.

File: perfkitbenchmarker/providers/aws/aws_network.py (hunk 2 above)
Comment (skschneider) [inline perfkitbenchmarker/providers/aws/aws_network.py]: Makes sense. It should only be created through AwsNetwork.

**Pull request merged**